### ISSUE TITLE
Should freeze_time accept lambdas or generators?

### ISSUE THREAD

**Issue by azmeuk** [spulec/freezegun#182]
Hi. I would love to see the following code work:
```python
import freezegun, datetime

dates = (datetime.datetime(year, 1, 1) for year in range(1, 2000))
with freezegun.freeze_time(lambda: next(dates)):
    print(datetime.datetime.now()) # would print the first of January "1
    print(datetime.datetime.now()) # would print the first of January "2
```
The idea is that in some test one may want that each `datetime.now` call return a different value he decided himself. This code is a simple example, but we can easily imagine lambdas more complicated.

What do you think? Would you accept pull requests for such a feature?

**Comment by rodude123** [comment]
I think it should accept both I have tried both and both work fine

**Comment by azmeuk** [comment]
@rodude123 The sample of code I provided raises this exception: `TypeError: freeze_time() expected None, a string, date instance, or datetime instance, but got type <type 'function'>.`

**Comment by spulec** [comment]
If it doesn't work, I'd be happy to accept a PR to make it work. It seems like a reasonable use case.

**Comment by spulec** [comment]
Fixed with #187
